File: ch06/test2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int fib(int n) {
    int a = 1;
    for (int i = n; i > 0; i--) {
        a = a * i;
    }
    return a;
}

int main(int argc, char *argv[]) {
    int n = atoi(argv[1]); // Example input
    int result = fib(n);
    printf("fib(%d) = %d\n", n, result);
    return 0;
}

File: ch06/test2.lua
function fib(n,k)
    local a = 1
    local b = 1
    for i = k*n, 1, -k do
        a = a + i
    end
    return a
end

fib(arg[1],arg[2])
-- Example usage
-- local n = 10000000 -- Example input
-- local result = fib(n)
-- print(string.format("fib(%d) = %d", n, result))
-- print(string.dump(fib))